src/api/util.ts

```diff
@@ -65,3 +72,19 @@ export function url(parts: TemplateStringsArray, ...params: any[]): string {
     result.push(parts[i]);
     return result.join("");
 }
+
+/**
+ * Merges two RequestInit objects, overriding the parameters of the left-hand
+ * side with the parameters of the right-hand side, except for the `headers`
+ * parameter, which is also merged separately.
+ */
+function mergeInit(a?: RequestInit, b?: RequestInit): RequestInit {
+    const headers = {};
+    if(a != null) {
+        Object.assign(headers, a.headers);
+    }
+    if(b != null) {
+        Object.assign(headers, b.headers);
+    }
+    return {...a, ...b, headers};
```

Comment:
What does the ... do?

Reply:
`{...a}` creates a new object, and copies over all properties from `a` to the new object. 

It resolves in left to right order, so in the case of `{...a, ...b, headers}` the values of `b` override the values of `a`, and finally `headers` is set.